test/basic-querying.test.js

```diff
@@ -654,6 +654,7 @@ describe('basic-querying', function() {
             if (err) done(err);
             users.should.have.property('length', 3);
             users[0].name.should.equal('John Lennon');
+            should(users[0].toObject()).be.a.function;
```

Comment:
This is weird. `users[0].toObject()` calls `toObject` function and then you are asserting that the retval of that function is a function again? Also I think that `be.a.function` is a no-op, you need to call the assertion - `be.a.function()`.

Since your intention is to verify that returned items are instances of the model class (and not just plain-data objects), maybe you can use `instanceof` check against a base model class?

```js
users[0].should.be.instanceOf(User);
```

(The `User` model is defined at the top of the test file and is accessible to all tests. In this particular test, you start by calling `User.find`, a method on that model.)

Reply:
Oops I meant to do `should(users[0].toObject).be.a.function()` 😅. Thank you for pointing it out and will use the `instanceOf` assertion which is exactly what I am looking for!

Reply:
> It's weird when I was debugging it, `User` was undefined even though like you said it's defined globally at the top.

I think `User` was undefined in the debugger because your callback was not accessing it and thus the variable was not captured in callback function's scope.